app/service/search_index.rb

```diff
@@ -9,35 +9,57 @@ def self.log_elasticsearch_warning(message)
     end
   end
 
-  def initialize(model_name:, id:)
+  def initialize(model_name:, id:, changed: nil)
     @model_name = model_name
     @id = id
+    @changed = changed
   end
 
   def add
-    if search_index_callbacks_enabled?
-      record = model_name.constantize.find(id)
-      record.__elasticsearch__.index_document
-    else
-      log_callback_warning
+    perform { record.__elasticsearch__.index_document }
+  end
+
+  def update
+    return add unless changed.any?
```

Comment:
Hmm, I think you're right. I had a lot of issues with the different callbacks / changes methods (which are different if you use `after_commit`, `after_update`, etc) and I might have forgotten to remove this. Gonna run a few more tests and see if I can remove it.